lib/reline/line_editor.rb

```diff
@@ -1131,23 +1131,17 @@ def just_move_cursor
     if Reline::Unicode::CSI_REGEXP.match?(prompt + line_to_render)
       @output.write "\e[0m" # clear character decorations
     end
-    visual_lines.each_with_index do |line, index|
+    visual_lines.each do |line|
       Reline::IOGate.move_cursor_column(0)
       if line.nil?
-        if calculate_width(visual_lines[index - 1], true) == Reline::IOGate.get_screen_size.last
```

Comment:
`visual_lines` is calculated by `Reline::Unicode.split_by_width`
```ruby
screen_width = 4
Reline::Unicode.split_by_width('helloworld', screen_width).first 
# => ["hell", nil, "owor", nil, "ld"]
irb(main):021> Reline::Unicode.split_by_width('Hello World!', screen_width).first 
# => ["Hell", nil, "o Wo", nil, "rld!", nil, ""]
Reline::Unicode.split_by_width('helloあwいoうrlえおd', screen_width).first 
#=> ["hell", nil, "oあw", nil, "いo", nil, "うrl", nil, "えお", nil, "d"]
```

Odd index value is always nil.
Normally, Even index value except last value satisfies `Reline::Unicode.calculate_width(_1) == screen_width`.
If there is a fullwidth character, text width might not be equal to screen_width.

The condition is always true for halfwidth chars.
I think condition is trying to check if `visual_lines[index - 1]` is the last element or not.
It's always true because `visual_lines[index]` is nil and `visual_lines[index + 1]` is the last element of visual_lines.
So this condition must be true for fullwidth chars too.